docs/rfcs/component-universal-telemetry.md

```diff
@@ -0,0 +1,119 @@
+# Auto-Instrumented Component Telemetry
+
+## Motivation
+
+The collector should be observable and this must naturally include observability of its pipeline components. It is understood that each _type_ (`filelog`, `batch`, etc) of component may emit telemetry describing its internal workings, and that these internally derived signals may vary greatly based on the concerns and maturity of each component. Naturally though, the collector should also describe the behavior of components using broadly normalized telemetry. A major challenge in pursuit is that there must be a clear mechanism by which such telemetry can be automatically captured. Therefore, this RFC is first and foremost a proposal for a _mechanism_. Then, based on what _can_ be captured by this mechanism, the RFC describes specific metrics, spans, and logs which can be broadly normalized.
+
+## Goals
+
+1. Articulate a mechanism which enables us to _automatically_ capture telemetry from _all pipeline components_.
+2. Define attributes that are (A) specific enough to describe individual component [_instances_](https://github.com/open-telemetry/opentelemetry-collector/issues/10534) and (B) consistent enough for correlation across signals.
+3. Define specific metrics for each kind of pipeline component.
+4. Define specific spans for processors and connectors.
```

Comment:
Good question. It's because the proposal is entirely based on the mechanism described, which is to insert instrumentation _in between_ pipeline components. In other words, there is a clear way to interact with data flowing both in and out of processors and connectors, but receivers and exporters we only have a generic way to interact with the data at one end.

All that said, I realized there is a flaw in my thinking about processors and connectors, because we cannot assume a direct relationship between data passed in and emitted from these components. We may be starting a lot of spans and not ending some, or ending the wrong ones. I will remove spans from this proposal for now.

Reply:
This is no longer in the RFC text, marking the comment as resolved